### PR TITLE
Add ability to yank inline

### PR DESCRIPTION
**Pull request by jgkamat**
See https://github.com/qutebrowser/qutebrowser/pull/1937, https://github.com/qutebrowser/qutebrowser/issues/2017, https://github.com/qutebrowser/qutebrowser/issues/1861, https://github.com/qutebrowser/qutebrowser/issues/4647, https://github.com/qutebrowser/qutebrowser/pull/3562.

I think this is a cleaner way to solve the issue than #1862, without breaking backwards compatibility.

FWIW, I don't think that adding expansions like `{title}` and `{domain}` should block PRs like that. There are more people affected by the lack of those domains, than the people that this change would break. Either way, we can now escape those by double-quoting the variables, so let's work on merging #1937 as well?

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/4651)

### CODE REVIEW

File: qutebrowser/browser/commands.py [hunk 1]
```diff
@@ -646,8 +646,11 @@ def _yank_url(self, what):
 
     @cmdutils.register(instance='command-dispatcher', scope='window')
     @cmdutils.argument('what', choices=['selection', 'url', 'pretty-url',
-                                        'title', 'domain', 'markdown'])
-    def yank(self, what='url', sel=False, keep=False, quiet=False):
+                                        'title', 'domain', 'markdown',
+                                        'inline'])
+    @cmdutils.argument('inline')
```
Comment (The-Compiler) [inline qutebrowser/browser/commands.py]: Not needed if you don't actually want to customize that argument in any way.

File: qutebrowser/browser/commands.py [hunk 2]
```diff
@@ -659,12 +662,19 @@ def yank(self, what='url', sel=False, keep=False, quiet=False):
                 - `domain`: The current scheme, domain, and port number.
                 - `selection`: The selection under the cursor.
                 - `markdown`: Yank title and URL in markdown format.
+                - `inline`: Yank the text contained in the 'inline' argument.
 
+            trailing: a trailer.
```
Comment (The-Compiler) [inline qutebrowser/browser/commands.py]: huh?

**Comment by The-Compiler** [comment]
Thanks for picking this up again! I think I agree with moving these forward even before looking at #2017 (which will probably still take a while).

Two questions remain:

- Should we deprecate `:yank markdown` in favor of something like `:yank inline [{title}]({url})` or so after this is implemented?
- Is "inline" really a good name for this? Does "yank an inline block" make sense? Why not "template", or maybe even just "text"?

**Comment by jgkamat** [comment]
Florian Bruhin writes:

> * Should we deprecate :yank markdown in favor of something like :yank inline [{title}]({url}) or so after this is implemented?

Not sure, I don't exactly like breaking backwards compatibility but this seems
like a trivial change. I'm fine either way. No idea how hard it would be, but
maybe we can provide completion for common formats for 'inline' to make it
easier to yank on the fly.

> * Is "inline" really a good name for this? Does "yank an inline block" make sense? Why not "template", or maybe even just "text"?

I couldn't think of a better one to be honest. I don't particularly like
'text' (could be confused with text on the document), and template sounds a
bit weird to me (it refers to the format of the thing being yanked, rather
than the thing being yanked itself). I do like 'immediate' though. I like
these names because it's obvious (to me at least) that the yank target is in
the command itself, unlike the other yank targets. Maybe 'next', 'following',
or 'args' would work too, but I like those less.

I'm fine with switching it to whatever you decide though, I don't care about
the name/wording as much as getting this feature in.

**Comment by The-Compiler** [comment]
Thanks @jgkamat, and thanks @blyxxyz for the original change!

I decided to deprecate `:yank markdown` in 05d052c0facc1d8803f07ede53835ddf07851df8.

**Pull request merged**